modules/data-entry/src/main/java/io/uhndata/cards/dataentry/internal/serialize/labels/AbstractResourceLabelProcessor.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.uhndata.cards.dataentry.internal.serialize.labels;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.osgi.service.component.annotations.Component;
+
+import io.uhndata.cards.serialize.spi.ResourceJsonProcessor;
+
+/**
+ * Fills in the clinic answer for a given visit information clinic node.
+ *
+ * @version $Id$
+ */
+@Component(immediate = true)
+public abstract class AbstractResourceLabelProcessor extends SimpleAnswerLabelProcessor implements ResourceJsonProcessor
+{
+    private static final String PROP_RESOURCE_LABEL = "labelProperty";
+
+    /**
+     * Given a resource question definition, find the resource property that should be used as the label for
+     * answers of that type, if it exists.
+     * @param question the Question node
+     * @return the property name as a string, or null if the question definition doesn't have the labelProperty
+     *     defined or its value is blank
+     */
+    protected String getLabelPropertyName(final Node question)
+    {
+        try {
+            if (question != null
+                && question.getProperty(PROP_RESOURCE_LABEL) != null
```

Comment:
`getProperty` throws an exception if the property is not there. Checking `!=null` doesn't make sense, either the call succeeds and it is not null, or it throws and goes to the catch block.

```suggestion
                && question.hasProperty(PROP_RESOURCE_LABEL)
```

Reply:
Fixed in 97c0e49 .